File: files/400-499/444/CH-444.py
import re
import pandas as pd

path = "400-499/444/CH-444 Replacement.xlsx"
input = pd.read_excel(path, usecols="B:D", nrows=7, skiprows=2)
test = pd.read_excel(path, usecols="F:H", nrows=7, skiprows=2)
test.columns = ["Date", "Product ID", "Total Sales"]


def replace_x(product_id):
    return re.sub(r"(?<!Y)X(?!Y)", "Y", product_id)


result = input.assign(**{"Product ID": input["Product ID"].map(replace_x)})
print(result.equals(test))
# True

File: files/400-499/444/CH-444.R
library(tidyverse)
library(readxl)

path <- "400-499/444/CH-444 Replacement.xlsx"
input <- read_excel(path, range = "B3:D9")
test <- read_excel(path, range = "F3:H9")

replace_x <- function(id) {
  str_replace_all(id, "(?<!Y)X(?!Y)", "Y")
}

result <- input %>%
  mutate(`Product ID` = map_chr(`Product ID`, replace_x))

all.equal(result, test)
# True
